.github/workflows/codeguru-scan.yml

```diff
@@ -0,0 +1,50 @@
+name: CodeGuru Review
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+      - feature/**
+
+jobs:
+  code-analysis:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      actions: read
+      contents: read
+      security-events: write
+      statuses: write
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 # This is a required field for CodeGuru
+    
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Comment:
Do we need some kind of approval to store these credentials externally?

Reply:
Tracked in security review doc, and have followed suggestions as per the Codeguru AWS Docs for configuration.